src/ui/molecules/switch/index.tsx

```diff
@@ -0,0 +1,188 @@
+  onChange: React.EventHandler<React.SyntheticEvent>;
+}
+
+const SwitchBase: React.FC<SwitchProps & Variant> = ({
+  className,
+  id,
+  isChecked,
+  isDisabled,
+  onChange,
+  variant = 'default',
+  ...p
+}) => {
+  const tabIndex = isDisabled ? -1 : 0;
+
+  const onKeyDown = React.useCallback(
+    (event: React.KeyboardEvent) => {
+      if (event.key === 'Enter') {
+        event.preventDefault();
+      }
+      const keyHandler = {
+        enter: (event: React.SyntheticEvent<Element, Event>) => {
+          onChange(event);
+        },
+      };
+
+      keyboardEventHandle({
+        event,
+        keyHandler,
+      });
+    },
+    [onChange],
+  );
+
+  return (
+    <div
+      className={className}
+      data-disabled={isDisabled}
+      data-variant={variant}
+      onKeyDown={onKeyDown}
+      tabIndex={tabIndex}
+    >
+      <label htmlFor={id}>
+        <input
+          checked={isChecked}
+          id={id}
+          onChange={onChange}
+          type="checkbox"
+          {...p}
+        />
+        <span data-checkbox>
+          <span />
+        </span>
+      </label>
+    </div>
+  );
+};
+
+export const Switch = styled(SwitchBase)`
+  --switcher-margin: calc(
+    var(--woly-switcher-height, 30px) / 2 -
+      var(--woly-switcher-tumbler-height, 24px) / 2
+  );
+  --switcher-toggle-margin: calc(
+    var(--woly-switcher-height, 30px) / 2 -
+      var(--woly-switcher-tumbler-height, 24px) / 2 - 1.5px
+  );
+  position: relative;
+  display: flex;
+  align-items: center;
+  outline: none;
+
+  & > div {
+    padding-right: 15px;
+  }
+
+  label > [data-checkbox] {
+    position: relative;
+    width: 57px;
+    height: 30px;
+    display: flex;
+  }
+
+  label > input {
+    display: none;
+    outline: none;
+  }
+
+  span[data-checkbox] > span {
+    position: absolute;
+
+    width: var(--woly-switcher-width, 57px);
+    height: var(--woly-switcher-height, 30px);
+
+    background-color: var(--woly-canvas, #c0c0c0);
+    border-color: var(--woly-canvas, #c0c0c0);
+    border-style: solid;
+    border-radius: var(--woly-rounding, 18px);
+    border-width: 0;
+    cursor: pointer;
+
+    &:before {
+      position: absolute;
+      border-width: 0;
+      width: var(--woly-switcher-tumbler-width, 24px);
+      height: var(--woly-switcher-tumbler-height, 24px);
+      margin: var(--switcher-margin, 3px);
+
+      background-color: var(--woly-switcher-tumbler-background-color, #ffffff);
+      border-color: var(--woly-switcher-tumbler-background-color, #ffffff);
+
+      border-radius: var(--woly-rounding, 50px);
+      cursor: pointer;
+      border-style: solid;
+      content: '';
+      :focus {
+        border-color: var(--woly-border, #b0a3f4);
+      }
+    }
+  }
+
+  input:checked + span[data-checkbox] > span {
+    background-color: var(--woly-background, #b0a3f4);
+    border-color: var(--woly-border-focus, var(--woly-background, #000000));
+  }
+
+  input:checked + span[data-checkbox] > span:before {
```

Comment:
можно просто написать  input:checked + span[data-checkbox] > span {
.....
&:before {....}
}

Reply:
а в чем будет разница?

Reply:
лишняя строка уйдет, зачем дублировать

Reply:
ага поняла про чо ты)

Reply:
```suggestion
  input:checked + [data-checkbox] > span:before {
```

мне кажется лучше убрать все эти span'ы

Второй селектор `span:before` вообще непонятно что выбирает. Какой span? Что он делает?

тут поможет вменяемый data-атрибут